### ISSUE TITLE
Fix `declaration-block-no-duplicate-properties` false negatives for `ignore: ['consecutive-duplicates-with-different-syntaxes']` and `<named-color>`

### ISSUE THREAD

**Issue by fireflysemantics** [stylelint/stylelint#8535]
### What minimal example or steps are needed to reproduce the bug?

Stylelint does not lint this.

```css
p {
 color: red;
 color:blue;
}
```

### What minimal configuration is needed to reproduce the bug?

```
{
  "extends": ["stylelint-config-standard"]
}
```

### How did you run Stylelint?

```
npx stylelint '**/*.{css,scss}'
```

### Which Stylelint-related dependencies are you using?

```
 "stylelint": "^16.18.0",

```

### What did you expect to happen?

It should have linted the second declaration.  [Also asked about it on SO](https://stackoverflow.com/questions/79563469/should-stylelint-flag-the-declaration-of-the-same-property-twice-within-a-code-b?noredirect=1#comment140314678_79563469)

### What actually happened?

It does not does not lint double declaration of color properties.

**Comment by jeddy3** [comment]
@fireflysemantics Thank you for sharing more of your thoughts.

The built-in rules balance different constraints and user needs. There is typically a tension between false negatives and false positives, and we strive to find the best compromise for each rule. We typically favour false negatives over false positives as the latter is more disruptive to users. You can read some discussions on how this was applied to this particular rule [here](https://github.com/stylelint/stylelint/issues/6632) and [here](https://github.com/stylelint/stylelint/issues/6796).

> [Also the stylelint documentation](https://stylelint.io/user-guide/rules/declaration-block-no-duplicate-properties/) has a lot of examples that claim that duplicate properties should be listed.

This is for the strictest setting. The built-in rules are strict by default and can be made more permissive using the options. When deciding what to turn on the standard config, we choose a point on that spectrum that caters to most users. In this case, we've chosen the option that tries to strike a balance between strictness and supporting some progressive enhancement practices.

You can make the rule stricter by using it without options:

```json
{
  "extends": ["stylelint-config-standard"],
  "rules": {
    "declaration-block-no-duplicate-properties": true
  }
}
```

Having said that, I think there's scope to discuss if the following is a false negative that we could address for the `ignore: ['consecutive-duplicates-with-different-syntaxes']` secondary option as the keywords are both [`<named-color>`](https://csstree.github.io/docs/syntax/#Type:named-color), which we maintain our [own reference](https://github.com/stylelint/stylelint/blob/10cc6158feccd5c1c030121f347a039a37b7bd06/lib/reference/keywords.mjs#L487) of:

```css
a {
    color: blue;
    color: red;
}
```

We'll need to decide if adding an exception for this one set of keywords is worth the overhead. I think it might be as it'll match the existing behaviour of colour formats, e.g.

```css
a {
  color: rgb(0 0 0);
  color: rgb(10 0 0);
}

b {
  color: #000;
  color: #FFF;
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLpANhAE7wpEDmARgBQAMK9tAlANxa4HGkU0CMjDVlgC+7TJTTs8EQiRQBiWkrbZps0vIBi2laOyYQAGhAAzAJb4YAOSQBbOIhgAPOwAdLAOjABnb0fAQmObkCCAYmCgo6CDOODCYUN7RpADaWJGR0d44AJ6W+GaYOAC0kEFm5MXZSAlIRFDR6SgAuiL+ZcGaxLZIOKEAVt6B-rCufojhmSDZeTAFRclRIPi9MNnRhk1ZufmFJR0VVTg1UHUNCEsrcesgbcJAA).

If there are no strong opinions either way, I can label the issue as a bug.

**Comment by romainmenke** [comment]
> We'll need to decide if adding an exception for this one set of keywords is worth the overhead. I think it might be as it'll match the existing behaviour of some other colour formats, e.g.

I think this could be fine.

The CSSWG is very unlikely to add new color keywords to the `namedColorsKeywords` set.
And support for this set is universal at this point.

In other words this is always a mistake:

```css
a {
  color: red;
  color: blue;
}
```

But we should not extend this same exception to other lists of keywords or to the set of system keywords as browser support typically varies in other sets.

-----

I do wonder how often this comes up in practice 🤔
I think it is fairly rare for people to use the CSS color keywords.
Most designs will use brand colors that can only be expressed in hex or `rgb()`.

And having a duplicate property of this kind doesn't affect the page render. It only adds a tiny bit of bloat to the CSS file.

But the effort to resolve this issue also seems fairly small :)

**Comment by jeddy3** [comment]
All good points. As the effort seems small, let's include it.

@fireflysemantics I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by sw1tch3roo** [comment]
Hi! I saw that the issue was still open so I’d like to work on this, let me know if that’s okay @jeddy3
